pkg/bucketaccessrequest/bucketaccessrequest.go

```diff
@@ -57,13 +58,24 @@ func (b *bucketAccessRequestListener) Add(ctx context.Context, obj *v1alpha1.Buc
 	return nil
 }
 
+// Update is called in response to a change to BucketAccessRequesst. At this point
```

Comment:
For [L48](https://github.com/kubernetes-sigs/container-object-storage-interface-controller/pull/29) do we want to validate that the existing BA is for this BAR?